packages/cdktf/lib/iterator.ts

```diff
@@ -0,0 +1,198 @@
+import { TerraformStack } from ".";
+import { ComplexList, ComplexMap } from "./complex-computed-list";
+import { Fn } from "./terraform-functions";
+import { propertyAccess, ref } from "./tfExpression";
+import { Lazy, Token } from "./tokens";
+
+export interface IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  _getForEachExpression(): any;
+}
+
+export abstract class Iterator implements IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  abstract _getForEachExpression(): any;
+
+  /**
+   * Creates a new iterator from a list
+   */
+  public static fromList(
+    list: string[] | number[] | ComplexList
+  ): ListIterator {
+    // TODO: this could return different iterators depending on the type of the list
```

Comment:
TODO left

Reply:
That one was on purpose. Should I just keep the comment without the todo prefix?